### PR TITLE
[docs] no hard-coding cuda

### PR DESCRIPTION
**Pull request by faaany**
## What does this PR do?
This PR adds xpu besides cuda and replaces the hard-coded cuda with device-agnostic device. 

@muellerzr

### CODE REVIEW

**Review by BenjaminBossan — APPROVED** [review]
LGTM.

File: docs/source/usage_guides/big_modeling.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ Now that the model is fully dispatched, you can perform inference.
 
 ```py
 input = torch.randn(2,3)
-input = input.to("cuda")
+input = input.to(model.device.type)
```
Comment (XuehaiPan) [inline docs/source/usage_guides/big_modeling.md:67]: I'm afraid the `model` does not always have a `device` member.

Comment (faaany) [reply docs/source/usage_guides/big_modeling.md:67]: also when the model is loaded with `load_checkpoint_and_dispatch`? I think when we consider the context, model should be a HF ModelClass, which will have device as an attribute. But pls correct me if I am wrong.

Comment (muellerzr) [reply docs/source/usage_guides/big_modeling.md:67]: We don't, esp as the prior sentence considers a regular pytorch model.

We can instead do one of two options:

1. Bring in the `PartialState` for this tutorial
2. Go a bit easier and use the model itself

(More a fan of 2 here)

```suggestion
device_type = next(iter(model.parameters())).device.type
input = input.to(device_type)
```

**Review by muellerzr — COMMENTED** [review]
Thanks! Let's make this a bit more agnostic as @XuehaiPan recommends :)

**Comment by faaany** [comment]
Thanks for the suggestion! Code Updated. @muellerzr

**Review by muellerzr — APPROVED** [review]
Thanks!

**Pull request merged**